operator/rediscontroller/webhook_test.go

```diff
@@ -0,0 +1,46 @@
+package rediscontroller
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+)
+
+func TestValidator_validateRedisSettings(t *testing.T) {
+	tests := map[string]struct {
+		givenSpec     string
+		expectedError string
+	}{
+		"EmptySettings":     {givenSpec: "", expectedError: ""},
+		"EmptyObject":       {givenSpec: "{}", expectedError: ""},
+		"Null":              {givenSpec: `null`, expectedError: ""},
+		"Float":             {givenSpec: `{"number":0.5}`, expectedError: ""},
+		"Zero":              {givenSpec: `{"number":0}`, expectedError: ""},
+		"Integer":           {givenSpec: `{"number":1}`, expectedError: ""},
+		"NegativeNumber":    {givenSpec: `{"number":-1.5}`, expectedError: ""},
```

Comment:
Maybe we can extract the core function to parse properties to its own function so that we only need to test this once?